### PR TITLE
Release module

### PR DESCRIPTION
**Pull request by Riccardo014**

### CODE REVIEW

File: apps/api/src/app/modules/project/dtov/release-create.dtov.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { ReleaseCreateDto } from '@api-interfaces';
+import { IsDate, IsNumber, IsOptional, IsString } from 'class-validator';
+
+export class ReleaseCreateDtoV implements ReleaseCreateDto {
+  @IsString()
```
Comment (RTsosssoldi) [inline apps/api/src/app/modules/project/dtov/release-create.dtov.ts:5]: IsUuuid

File: apps/api/src/app/modules/project/dtov/release-extra-hours-create.dtov.ts [hunk 2]
```diff
@@ -12,4 +12,7 @@ export class ReleaseExtraHoursCreateDtoV implements ReleaseExtraHoursCreateDto {
   @IsString()
   referent: string;
 
+  @IsString()
```
Comment (RTsosssoldi) [inline apps/api/src/app/modules/project/dtov/release-extra-hours-create.dtov.ts:15]: isUUid

File: apps/api/src/app/modules/project/dtov/release-extra-hours-create.dtov.ts [hunk 3]
```diff
@@ -12,4 +12,7 @@ export class ReleaseExtraHoursCreateDtoV implements ReleaseExtraHoursCreateDto {
   @IsString()
```
Comment (RTsosssoldi) [inline apps/api/src/app/modules/project/dtov/release-extra-hours-create.dtov.ts:12]: IsOptional

File: apps/api/src/app/modules/project/dtov/release-extra-hours-update.dtov.ts [hunk 4]
```diff
@@ -15,4 +15,8 @@ export class ReleaseExtraHoursUpdateDtoV implements ReleaseExtraHoursUpdateDto {
   @IsString()
   referent: string;
   
+  @IsOptional()
+  @IsString()
```
Comment (RTsosssoldi) [inline apps/api/src/app/modules/project/dtov/release-extra-hours-update.dtov.ts:19]: IsUUID

File: apps/api/src/app/modules/project/dtov/release-update.dtov.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { ReleaseUpdateDto } from '@api-interfaces';
+import { IsDate, IsNumber, IsOptional, IsString } from 'class-validator';
+
+export class ReleaseUpdateDtoV implements ReleaseUpdateDto {
+  @IsOptional() 
+  @IsString()
```
Comment (RTsosssoldi) [inline apps/api/src/app/modules/project/dtov/release-update.dtov.ts:6]: IsUUId

File: apps/api/src/app/modules/project/entities/project.entity.ts [hunk 6]
```diff
@@ -8,7 +8,7 @@ export class Project extends BaseEntityTemplate{
   @Column()
   name: string;
 
-  @ManyToOne(() => Customer)
+  @ManyToOne(() => Customer, { onDelete: 'CASCADE' })
```
Comment (RTsosssoldi) [inline apps/api/src/app/modules/project/entities/project.entity.ts:11]: onDelete metterei SET NULL, perchè voglio tenere comuqnue lo storico dei progetti se cancello un customer

File: apps/api/src/app/modules/user/entities/user.entity.ts [hunk 7]
```diff
@@ -14,7 +14,7 @@ export class User extends BaseEntityTemplate{
 
   _password: string | null;
 
-  @ManyToOne(() => Role)
+  @ManyToOne(() => Role, { onDelete: 'CASCADE' })
```
Comment (RTsosssoldi) [inline apps/api/src/app/modules/user/entities/user.entity.ts:17]: Anche qui set null, non deve poterelo fare. Cioè se voglio cancellare un ruolo devo prima spostare gli utenti di quel ruolo su un altro

**Review by RTsosssoldi — CHANGES_REQUESTED** [review]

File: apps/api/src/app/modules/project/dtov/release-create.dtov.ts (hunk 1 above)
Comment (Riccardo014) [reply apps/api/src/app/modules/project/dtov/release-create.dtov.ts]: https://github.com/RainTonic/controllo_ore_x/pull/8/commits/64485332e0bfd3e3500110179093b0bb494d7f1e

File: apps/api/src/app/modules/project/dtov/release-extra-hours-create.dtov.ts (hunk 2 above)
Comment (Riccardo014) [reply apps/api/src/app/modules/project/dtov/release-extra-hours-create.dtov.ts]: https://github.com/RainTonic/controllo_ore_x/pull/8/commits/64485332e0bfd3e3500110179093b0bb494d7f1e

File: apps/api/src/app/modules/project/dtov/release-extra-hours-create.dtov.ts (hunk 3 above)
Comment (Riccardo014) [reply apps/api/src/app/modules/project/dtov/release-extra-hours-create.dtov.ts:13]: https://github.com/RainTonic/controllo_ore_x/pull/8/commits/64485332e0bfd3e3500110179093b0bb494d7f1e

File: apps/api/src/app/modules/project/dtov/release-extra-hours-update.dtov.ts (hunk 4 above)
Comment (Riccardo014) [reply apps/api/src/app/modules/project/dtov/release-extra-hours-update.dtov.ts]: https://github.com/RainTonic/controllo_ore_x/pull/8/commits/64485332e0bfd3e3500110179093b0bb494d7f1e

File: apps/api/src/app/modules/project/dtov/release-update.dtov.ts (hunk 5 above)
Comment (Riccardo014) [reply apps/api/src/app/modules/project/dtov/release-update.dtov.ts]: https://github.com/RainTonic/controllo_ore_x/pull/8/commits/64485332e0bfd3e3500110179093b0bb494d7f1e

File: apps/api/src/app/modules/user/entities/user.entity.ts (hunk 7 above)
Comment (Riccardo014) [reply apps/api/src/app/modules/user/entities/user.entity.ts]: https://github.com/RainTonic/controllo_ore_x/pull/8/commits/7e7518b8b363be97eef9df29586275ff541e9f3d

File: apps/api/src/app/modules/project/entities/project.entity.ts (hunk 6 above)
Comment (Riccardo014) [reply apps/api/src/app/modules/project/entities/project.entity.ts]: ****

**Review by RTsosssoldi — APPROVED** [review]

**Pull request merged**